lib/logger.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var merge = require('lodash.merge');
+var util = require('util');
+var winston = require('winston');
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger.log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} options - Configuration options.
+ * @return {Object} An instance of Winston.
+ */
+exports.logger = function (options) {
+  'use strict';
+
+  options = getOptions(options);
+
+  var exitOnError = options.logger.exitOnError;
+  var level = options.logger.level;
+  var logFile = options.logger.logFile;
+  var silent = options.logger.silent;
+  var transports = [
+    new winston.transports.Console({
```

Comment:
@e2tha-e 
cc: @rebmullin @sergesemashko @conortm @gvorbeck @beynya @nikkiana

No specific problem, just looking at our history. The impetus of this entire system (the pattern library) is solving the fact that we always move on to something new. Always. The pattern library was born because we need our html (and it's functionality and required styling) to be able to used in any system and also not be locked into any system. We also needed a way for individual patterns to be portable and upgradeable and version-able.

I continued this thinking into the architecture of this app.

## gulp tasks
The gulp tasks can be pulled in and configured to be used without the need to import that task's modules and also ship with configs. *Obviously* gulp tasks can be small things that just live in a gulpfile. But, by creating them as separate modules we can then set them up to have the specific configurations we need and import them into gulpfile.js for use.

This helps us to:

* stop copy n paste
    * if we didn't go this route, each time we want to use the same gulp task we're back to copy n paste.
* keep package management outside of app
    * by making pattern-library-utilities an `npm import` of any pattern library, we remove the need for the pattern library to care about what version of which npm module is currently the correct one

I realize that some of the included tasks are very pattern-library specific. But others are not, ghPages and doxx could be used any any gulp-based project. 

## avoiding tons of functions in one file

A typical set up for node modules is to create a long utility file with all kinds of little (and big) functions that do all kinds of things. I wanted to break this out for documentation and portability.

For instance, look at your `exports.inspect` function below. Very handy and very useful. We could use that by importing this repo and referencing it. Probably too small a function for such a purpose...but maybe not? We'd never have to write `depth = depth ? depth : 2; util.inspect(obj, showHidden, depth)` again...nor would the project we're working in need to pay attention to the changes in npm's util.

## an original concept of splitting the repository out

When we first started working on the pattern library system I was working with @ericduran to make the patterns automatically create their own respositories via jenkins. This was copying the model that the [symfony org](https://github.com/symfony) used to split out all their scripts so that an end user can use only one part of the system without having to pull in the entire repository and all it's scripts.

## general usage outside this system

What if we had a drop-in solution for the next node app our group creates for logging? converting yaml/twig/sass files? a system configuration?

There's been a bunch of reasons that I wanted these available, but it's really just the future. Right now we build something and move on a build the next thing. We transfer our collective knowledge to the new system, but I think it's time we kept our code too.